pandas/tests/frame/test_constructors.py

```diff
@@ -735,15 +776,15 @@ def test_constructor_corner(self):
 
         # does not error but ends up float
         df = DataFrame(index=lrange(10), columns=['a', 'b'], dtype=int)
-        assert df.values.dtype == np.object_
+        assert df.values.dtype == np.dtype('float64')
```

Comment:
why is this changing?

Reply:
Because it was wrong: an ``int`` should not upcast to ``object`` (the passed dtype is currently not considered). issue? whatsnew?

Reply:
hmm, yeah this looks suspect. I would make a new issue for this

Reply:
#19646